bigquery/schema.go

```diff
@@ -226,6 +226,10 @@ const (
 	// BigNumericFieldType is a numeric field type that supports values of larger precision
 	// and scale than the NumericFieldType.
 	BigNumericFieldType FieldType = "BIGNUMERIC"
+	// IntervalFieldType is a representation of a duration or amount of time.
+	//
+	// This type is in preview.  It is EXPERIMENTAL and subject to change or removal without notice.
```

Comment:
Is this sufficient to give you enough wiggle room to return something other than string in the future?

Reply:
I believe it is; users read data using custom marshaling into their own data structures or for the dynamic case use a type switch.  When we have a new go type for handling interval values we can augment the docs to indicate such.